pallets/services-payment/src/lib.rs

```diff
@@ -332,8 +368,12 @@ impl<T: Config> Pallet<T> {
                 WithdrawReasons::FEE,
                 ExistenceRequirement::AllowDeath,
             ) {
-                // Burn for now, we might be able to pass something to do with this
-                drop(imbalance);
+                if let Some(address) = RefundAddress::<T>::take(para_id) {
```

Comment:
This only removes it if the `parachain_tank_balance` is not zero, if it is zero it will not be removed

Reply:
Added a take instead of get which should remove it

Reply:
Yes but this take is inside an if condition, which may not always be true

Reply:
yeah you are right, in case of credits being used only this might not hit. I will modify and add a test accordingly